operator/pkg/model/translation/envoy_virtual_host.go

```diff
@@ -348,11 +445,18 @@ func getRouteMatch(hostnames []string, hostNameSuffixMatch bool, pathMatch model
 		}
 	}
 	if pathMatch.Prefix != "" {
-		return &envoy_config_route_v3.RouteMatch{
-			PathSpecifier: &envoy_config_route_v3.RouteMatch_SafeRegex{
-				SafeRegex: &envoy_type_matcher_v3.RegexMatcher{
-					Regex: getMatchingPrefixRegex(pathMatch.Prefix),
+		if pathMatch.Prefix == "/" {
```

Comment:
instead of introducing the nested `if`-statement I would refactor the whole method from the multiple `ifs` (with early returns) to a switch statement of the following form.

```
switch {
case pathMatch.Exact != "":
  return ...
case pathMatch.Prefix == "/":
  return ...
case pathMatch.Prefix != "":
  return ...
case pathMatch.Regex != "":
  return ...
default:
  return ...
}
```

Reply:
ah nice, I didn't know that switch {} is allowed :face_exhaling: